dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/SAMLHelper.java

```diff
@@ -0,0 +1,473 @@
+import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.cms.factories.PublicEncryptionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.util.ActivityLogger;
+import com.dotmarketing.util.AdminLogger;
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.SecurityLogger;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.portal.model.Company;
+import com.liferay.portal.model.User;
+import org.apache.commons.lang.StringUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static com.dotmarketing.util.UtilMethods.isSet;
+
+/**
+ * SAML Helper for the Endpoints
+ * @author jsanca
+ */
+public class SAMLHelper {
+
+    private final HostWebAPI hostWebAPI;
+    private final UserAPI    userAPI;
+    private final RoleAPI    roleAPI;
+    private final SamlAuthenticationService  samlAuthenticationService;
+
+    public SAMLHelper(final SamlAuthenticationService samlAuthenticationService) {
+
+        this.userAPI      = APILocator.getUserAPI();
+        this.roleAPI      = APILocator.getRoleAPI();
+        this.hostWebAPI   = WebAPILocator.getHostWebAPI();
+        this.samlAuthenticationService = samlAuthenticationService;
+    }
+
+    // Gets the attributes from the Assertion, based on the attributes
+    // see if the user exists return it from the dotCMS records, if does not
+    // exist then, tries to create it.
+    // the existing or created user, will be updated the roles if they present
+    // on the assertion.
+    protected User resolveUser(final Attributes attributes,
+                             final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        User user       = null;
+        User systemUser = null;
+        try {
+
+            Logger.debug(this, ()-> "Validating user - " + attributes);
+
+            systemUser             = this.userAPI.getSystemUser();
+            final Company company  = APILocator.getCompanyAPI().getDefaultCompany();
+            final String  authType = company.getAuthType();
+            user                   = Company.AUTH_TYPE_ID.equals(authType)?
+                    this.userAPI.loadUserById(this.samlAuthenticationService.getValue(attributes.getNameID()),      systemUser, false):
+                    this.userAPI.loadByUserByEmail(this.samlAuthenticationService.getValue(attributes.getNameID()), systemUser, false);
+        } catch (NoSuchUserException e) {
+
+            Logger.error(this, "No user matches ID '" +
+                    this.samlAuthenticationService.getValue(attributes.getNameID()) + "'. Creating one...", e);
+            user = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5956529)